### DISCUSSION TITLE
Update other field in database from Tables ToggleColumn

### DISCUSSION THREAD

**Discussion by valpuia** [filamentphp/filament#6277] [Help] [answered]
I have a table where user can toggle Status (0,1) without going to page/modal.

After they updated the status, I want to store `last_updated_by` in that particular data. _(store auth()->id() inside table)_

Is there any work around to update another field based on table toggle (or select, checkbox etc)

P.S: I've tried to use `Observer` but app is crashing.

**Comment by valpuia** [comment]
Can anyone help me.. thank you

**Reply by danharrin** [reply]
An observer is the correct approach, what is crashing

**Reply by valpuia** [reply]
In observer, all data is updated in database as expected.. but the `toggle` livewire (ajax) request is always in Pending.. cannot go/do anything.. the livewire request cannot get 204

**Reply by danharrin** [reply]
I don't understand. Send the code you tried

**Reply by valpuia** [reply]
In observer:

```
public function updated(Currency $currency): void
{
    $currency->update([
        'updated_by' => auth()->id(),
    ]);
}
```

In EventServiceProvider

```
public function boot(): void
{
    Currency::observe(CurrencyObserver::class);
}
```

Filament Table 

```
// this is toggle in table
Tables\Columns\ToggleColumn::make('is_active')
    ->label(__('message.active'))
    ->alignCenter()
    ->onIcon('heroicon-o-check')
    ->offIcon('heroicon-o-x')
    ->onColor('success')
    ->offColor('primary'),
```

When toggle currency `active` then the below request(image) cannot execute, always pending

![Screenshot from 2023-04-26 09-08-28](https://user-images.githubusercontent.com/28250303/234448331-81b4e3ce-475e-4c1f-86cd-e47c05085003.png)


After restart Apache as well as host and check again, then table is updated as expected.. If I don't use observer then this request is executed easily, but didn't update `updated_by` column

**Reply by danharrin** [reply]
Instead of updated(), try updating() and just `->set = ` the property instead of fully updating the model

**Reply by valpuia** [reply, marked as answer]
Yes, I've change based on your suggestion like below and it's working

```
public function updating(Currency $currency): void
{
    $currency->updated_by = auth()->id();
}
```
